Adding receipts to expenses

viktorcsimma opened the pull request:
Closes #331. Receipts are required for expenses; they can also be downloaded from the list, but they cannot be modified.

viktorcsimma committed:
Altering database and controllers; adding field to upload (but we cannot modify yet)

viktorcsimma committed:
Receipts can now be downloaded

viktorcsimma committed:
Receipts now get deleted along with transactions

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in 4ee3ecb according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/356

kdmnk commented on resources/views/utils/checkout/add-expense.blade.php:
```diff
@@ -33,6 +33,8 @@
                 <x-input.select m=6 l=6 id="payer" text="Fizető" :elements="\App\Models\User::collegists()" default="{{user()->id}}" :formatter="function($user) { return $user->uniqueName; }" />
                 <x-input.checkbox m=6 l=6 id="paid" checked text="A tartozás kifizetésre került"/>
                 @endcan
+                <x-input.file s="12" xl="8" id="receipt" style="margin-top:auto" accept=".pdf,.jpg,.png,.jpeg,.gif"
```
Only one question: why xl=8? I'd just delete it.

kdmnk commented on app/Utils/CheckoutHandler.php:
```diff
@@ -209,6 +212,9 @@ public function addExpense(Request $request): RedirectResponse
             'paid_at'           => $paid ? Carbon::now() : null,
         ]);
 
+        $path = $request->file('receipt')->store('receipts');
```
Wrap these and the Transaction::create in a DB::transaction(...), so if either of the logic fails it backtracks. Same with deletion.

viktorcsimma committed:
Organizing into DB transactions; fixing UI

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in 61f6180 according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/356

viktorcsimma commented:
Done:)

kdmnk reviewed (approved):
Codewise it's fine, you can test it live :P

One comment though: you can put the mail sending into the transaction as well. Now if that fails, the user gets an error message, yet the transactions are created properly. Depends how we want it.

viktorcsimma commented:
Hm... I have arguments both for and against it. Such a situation would be confusing; but then if anything happens with the mail function, it does not paralyse the function itself.

I think I'm going to merge it now; thank you:)
